plugins/modules/ipa_getkeytab.py

```diff
@@ -0,0 +1,250 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2024 Alexander Bakanovskii <skottttt228@gmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: ipa_getkeytab
+short_description: Manage keytab file in FreeIPA
+description:
+  - Manage keytab file with ipa-getkeytab utulity.
+author: "Alexander Bakanovskii (@abakanovskii)"
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: none
+options:
+  path:
+    description:
+      - The base path where to put generated keytab file.
+    type: path
+    aliases: ["keytab"]
+    required: true
+  principal:
+    description:
+      - The non-realm part of the full principal name.
+    type: str
+    required: true
+  ipa_server:
+    description:
+      - The IPA server to retrieve the keytab from (FQDN).
+    type: str
+    aliases: ["server"]
+  ldap_uri:
+    description:
+      - LDAP URI. If V(ldap://) is specified, STARTTLS is initiated by default.
+      - Can not be used with the O(ipa_server) option.
+    type: str
+  bind_dn:
+    description:
+      - The LDAP DN to bind as when retrieving a keytab without Kerberos credentials.
+      - Generally used with the O(bind_pw) option.
+    type: str
+  bind_pw:
+    description:
+      - The LDAP password to use when not binding with Kerberos.
+    type: str
+  password:
+    description:
+      - Use this password for the key instead of one randomly generated.
+    type: str
+  ca_certificate:
+    description:
+      - The path to the IPA CA certificate used to validate LDAPS/STARTTLS connections.
+    type: path
+    aliases: ["ca_cert"]
+  sasl_mech:
+    description:
+      - SASL mechanism to use if O(bind_dn) and O(bind_pw) are not specified.
+    choices: ["GSSAPI", "EXTERNAL"]
+    type: str
+  retrieve_mode:
+    description:
+      - Retrieve an existing key from the server instead of generating a new one.
+      - This is incompatible with the O(password), and will work only against a IPA server more recent than version 3.3.
+      - The user requesting the keytab must have access to the keys for this operation to succeed.
+      - Be aware that if set V(true), a new keytab will be generated.
+      - This invalidates all previously retrieved keytabs for this service principal.
+    type: bool
+  encryption_types:
+    description:
+      - The list of encryption types to use to generate keys.
+      - It will use local client defaults if not provided.
+      - Valid values depend on the Kerberos library version and configuration.
```

Comment:
Is there a minimum level for the Kerberos library? In any case, the library (with or without min version) should be cited in the requirements, for completeness.

Reply:
Maybe add a link to those man pages (in one of those sites that publish them)? Or some other relevant reference.

Reply:
It seems that there are not much trusted resources with man page of `ipa-getkeytab`, there are source file for that (https://github.com/freeipa/freeipa/blob/master/client/man/ipa-getkeytab.1) but other than that I found Ubuntu docs https://manpages.ubuntu.com/manpages/jammy/man1/ipa-getkeytab.1.html
What do you think? Should I add that link in the full description of a module?

Reply:
Yup, one of those (whichever you prefer). https://linux.die.net/man/1/ipa-getkeytab also appears out of google, just as an example.
Anyways, if you prefer not to add that to the description, maybe add it to the notes section.
Come to think of it, I will review all my modules (that call commands) to add links to the respective man pages. :-)

Reply:
Agreed :) I used ubuntu doc as linux.die.net don't have all options documented